### ISSUE TITLE
GPUParticlesCollisionHeightField3D is broken in 4.4dev6 and 4.4dev7

### ISSUE THREAD

**Issue by how2Bboss** [godotengine/godot#100793]
### Tested versions

Reproducible in 4.4dev6 official, and 4.4dev7 official
Not Reproducible in 4.4dev5 official

### System information

MacOS 15.2 - Apple M2 Pro - 16GB Memory - 4.4dev6 official - Rendering backend: Metal

### Issue description

GPUParticlesCollisionHeightField3D works fine in 4.4dev5, but either doesn't update or particles dont respond to it in 4.4dev6-7

### Steps to reproduce

Create a new 3d scene

Place a GPUParticles3D node and give it a ParticleProcessMaterial with collision either on **Rigid** or **Hide On Contact**
Give the particles a mesh so they are visible

Place a new MeshInstance3D node and give it any type of mesh, in my example i used a sphere

Finally place a GPUParticlesCollisionHeightField3D node and move it around so it updates its height field

You may need to move the GPUParticles node as that gets detected in the height field.

In 4.4dev5, this should work as intended, where the particles slide off or disappear when hitting the mesh
In 4.4dev6-7, the particles pass straight through the mesh

4.4dev5
![Image](https://github.com/user-attachments/assets/7c286b23-6af6-4552-a65f-ee7ed92344a9)

4.4dev6-7
![Image](https://github.com/user-attachments/assets/cfc99d99-7db0-4e6b-8f6f-edb0af2bf3b2)

### Minimal reproduction project (MRP)

[Archive.zip](https://github.com/user-attachments/files/18241769/Archive.zip)

**Comment by HerrSersch** [comment]
I bisected this, it looks like [6d5ac8f7ef4a3ddaf50720ab473b9dffece21674] has introduced this bug.

**Comment by Calinou** [comment]
I can't reproduce this on 4.4.beta 7b1ed520b (Linux, GeForce RTX 4090 with NVIDIA 565.77).

![Image](https://github.com/user-attachments/assets/4ad342aa-29d5-4a7c-b667-c53bbb37884f)

This is likely a GPU-specific issue, although I remember SDF collision being broken on AMD + Windows on https://github.com/godotengine/tps-demo (even in 4.3.stable) while it works fine on NVIDIA.

**Comment by DarioSamo** [comment]
Unable to reproduce.

Have you been able to reproduce this issue using MoltenVK instead?

**Comment by how2Bboss** [comment]
Yes, it seems to still be broken in 4.4 beta 1 using Metal or MoltenVK.

**Comment by DarioSamo** [comment]
I'll tag it as macos only for now as I've seen no reports of the issue elsewhere. I feel the bisect to the particular commit might just reveal there's some handling that these GPUs don't seem to like of how this effect works, but I'm not familiar with all the steps involved yet to say how it could affect it.

**Comment by how2Bboss** [comment]
It might be a good idea to check other things related to GPU particles. Collision shapes work fine, but I haven’t tested the Gpu particle sdf (or whatever it’s called). I can test that stuff later today

**Comment by DarioSamo** [comment]
I can confirm it doesn't work on an Apple M3 either on either Metal or Vulkan.

**Comment by how2Bboss** [comment]
Cool. I’ll check when I’m home

**Comment by how2Bboss** [comment]
Yup. Seems like it’s fixed in 4.4 beta 2 👍
